config/config.go

```diff
@@ -66,21 +55,90 @@ var defaultConfig = Config{
 }
 
 func LoadConfig(file_path string) (*Config, error) {
+	config := &Config{}
+
+	if file_path == "" {
+		file_path = "./config.yaml"
+	}
+
 	content, e := os.ReadFile(file_path)
 	if e != nil {
-		return nil, e
+		return config, e
 	}
 
-	var config Config
-
-	e = yaml.Unmarshal(content, &config)
-	if e != nil {
-		return nil, e
+	if err := yaml.Unmarshal(content, &config); err != nil {
+		return config, err
 	}
 
 	if config.Port == "" {
 		config.Port = defaultConfig.Port
 	}
 
-	return &config, nil
+	if err := config.validate(); err != nil {
+		return config, err
+	}
+
+	return config, nil
+}
+
+func (c *Config) validate() error {
+	endpoints := []string{}
+	for i, source := range c.Sources {
+		// Ensure that the endpoint is unique
+		if slices.Contains(endpoints, source.EndPoint) {
+			return fmt.Errorf(".Source.%d: EndPoint is not unique", i)
+		}
+		endpoints = append(endpoints, source.EndPoint)
+
+		if err := source.validate(); err != nil {
+			return fmt.Errorf(".Source.%d: %s", i, err)
+		}
+	}
+
+	return nil
+}
+
+type Source struct {
+	EndPoint  string       `yaml:"end_point"`
+	Heartbeat int          `yaml:"heartbeat"`
+	Name      string       `yaml:"xwr_name"`
+	Info      []SourceInfo `yaml:"info"`
+}
+
+func (c *Source) validate() error {
```

Comment:
Can we validate Heartbeat is greater than 0 as well?

Name and EndPoint should be fine being an empty string I think.